cranelift/codegen/src/isa/x64/abi.rs

```diff
@@ -430,6 +465,15 @@ impl ABIMachineSpec for X64ABIMachineSpec {
             ));
         }
 
+        // If this is Baldrdash-2020, restore the TLS reg.
+        if call_conv == isa::CallConv::Baldrdash2020 {
+            insts.push(Inst::mov64_m_r(
+                Amode::imm_reg(16, regs::rbp()),
```

Comment:
Here (and in the aarch64 code), can you use a const value to represent the offset to the TLS? (Maybe precise in the comment "caller" TLS, too?)

A few questions for my own understanding of this solution compared to others:

- would it make sense to put this responsibility on the caller's side, just after the call site, in the baldrdash glue? It's already done for calls to imports, as far as i can tell: https://searchfox.org/mozilla-central/source/js/src/wasm/cranelift/src/wasm2clif.rs#1012
- could we mark the TLS reg as part of the callee-saved registers in the Baldrdash calling convention, and not do anything particular here for this register?

Reply:
Done, thanks! A small point, but it's actually the *callee* TLS that we restore; my reading of the ABI document is that we're supposed to leave the TLS register value unchanged, i.e., it must be the same at exit as it is at entry, so that would be the callee value.

> would it make sense to put this responsibility on the caller's side

I thought about this too, but I don't think this follows the ABI strictly, as (given the above) we must do the restoration before the return.

> could we mark the TLS reg as part of the callee-saved registers

I considered this too, but tweaking the `RealRegUniverse` construction to work properly with *two* optionally-allocatable registers (the pinned register and the TLS register) proved a bit too complex; the indices for registers would have to be dependent on the flags/calling convention. If we refactor this as we discussed a few days ago, then we might want to reconsider, but for now, it seems simpler just to restore the register.